geth/process.py

```diff
@@ -180,14 +199,15 @@ def wait_for_ipc(self, timeout=0):
                 _timeout.check()
 
     @property
-    def is_dag_generated(self):
+    def is_dag_generated(self) -> bool:
         return is_dag_generated()
 
     @property
-    def is_mining(self):
-        return self.geth_kwargs.get("mine", False)
+    def is_mining(self) -> bool:
+        mine_bool = cast(bool, self.geth_kwargs.get("mine", False))
```

Comment:
`self.geth_kwargs` is an instance of the `GethKwargs`, which inherits from `BaseModel` and provides access to its attributes as class attributes rather than as dictionary elements. So, we can directly access the `mine` instead of using the `get`

```suggestion
        mine_bool = cast(bool, self.geth_kwargs.mine
```